rabbitmq/assets/configuration/spec.yaml

```diff
@@ -182,3 +182,49 @@ files:
         - type: multi_line
           name: logs_starts_with_equal_sign
           pattern: "="
+
+- name: auto_conf.yaml
+  options:
+  - template: ad_identifiers
+    overrides:
+      value.example:
+        - rabbitmq
+  - template: init_config
+    options: []
+  - template: instances
+    options:
+    - name: prometheus_plugin
+      description: |
+        Settings for talking to the Prometheus plugin of RabbitMQ.
+        Specify this section to trigger metric collection from the Prometheus plugin endpoints.
+      enabled: true
+      options:
+        - name: url
+          enabled: true
+          required: true
+          description: |
+            Base URL for OpenMetrics endpoints.
+          value:
+            type: string
+            example: 'http://%%host%%:15692'
+        - name: include_aggregated_endpoint
+          description: |
+            Whether or not to include metrics from the aggregated endpoint (/metrics).
+
+            Note: This option must be set to true when 'unaggregated_endpoint' is unspecified.
+          value:
+            type: boolean
+            example: true
+        - name: unaggregated_endpoint
+          description: |
+            Which  unaggregated endpoint to use.
+            The choices are:
+              - If the field is left unspecified, no endpoint is used.
+              - "detailed" means the /metrics/detailed endpoint is used.
+                You can also provide a query for this option, exactly as if you were making an HTTP request
+                to this endpoint. For example, `detailed?family=queue_consumer_count`.
+                You can specify this and the aggregated endpoint. In that case any duplicate metrics will
+                be taken from this endpoint and the corresponding aggregated ones dropped.
+          value:
+            type: string
+            example: "detailed"
```

Comment:
You could specify the possible values using the `enum` attribute. You have one example here https://github.com/DataDog/integrations-core/blob/0284eaf179a8f780920d23e8b6a02fd95afc1388/teradata/assets/configuration/spec.yaml#L127-L132

This way, the value will be validated automatically (for py > 3)

Maybe adding this in a follow-up PR which will also update the `spec.yaml` at the same time would be nice, what do you think?

Reply:
Hmm, do you know of a way to make `enum` support arbitrary queries from the user?

Reply:
ah no, we would need to have an extra parameter for the query itself, which is not really needed and would make it less easy to configure. LEt's keep it as it is!

Reply:
oki-doki!